controllers/suite_test.go

```diff
@@ -65,9 +69,23 @@ var _ = BeforeSuite(func() {
 	ctx, cancel = context.WithCancel(context.TODO())
 
 	By("bootstrapping test environment")
+	k3sContainer, err := k3s.RunContainer(ctx,
+		testcontainers.WithImage("docker.io/rancher/k3s:v1.27.1-k3s1"),
+	)
+	Expect(err).NotTo(HaveOccurred())
+
+	kubeConfigYaml, err := k3sContainer.GetKubeConfig(ctx)
+	Expect(err).NotTo(HaveOccurred())
+
+	restcfg, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigYaml)
+	Expect(err).NotTo(HaveOccurred())
+
+	trueValue := true
 	testEnv = &envtest.Environment{
 		CRDDirectoryPaths:     []string{filepath.Join("..", "config", "crd", "bases")},
 		ErrorIfCRDPathMissing: true,
+		Config:                restcfg,
+		UseExistingCluster:    &trueValue,
```

Comment:
Is this means that now we are running the tests in a cluster that actually reconcile resources (e.g. pod)? Am I understanding correctly?

Reply:
Yes!